var/spack/repos/builtin/packages/elsi/package.py

```diff
@@ -33,27 +33,51 @@ class Elsi(CMakePackage):
     variant("enable_sips", default=False, description="Enable SLEPc-SIPs support")
     variant("use_external_elpa", default=False, description="Build ELPA using SPACK")
     variant("use_external_ntpoly", default=False, description="Build NTPoly using SPACK")
-    variant("use_external_omm", default=False, description="Use external libOMM and MatrixSwitch")
     variant("use_external_superlu", default=False, description="Use external SuperLU DIST")
     variant(
         "use_mpi_iallgather", default=True, description="Use non-blocking collective MPI functions"
     )
+    variant("tests", default=False, description="Enable Fortran tests")
+    variant("tests_c", default=False, description="Enable C tests")
```

Comment:
isn't better to use the `--test` flag, by removing these variants and enabling testing conditionally on `self.run_tests`?

Reply:
Mabye I'm missing something, but these variants are to ensure the tests are built by CMake (which they are not by default).

Reply:
IMHO instead of using variants to enable them, I think you might use `self.run_tests` to enable them or not.

See some examples https://github.com/search?q=repo%3Aspack%2Fspack+self.run_tests&type=code

Reply:
Thanks for the suggestion!